### DISCUSSION TITLE
How to override some snippets?

### DISCUSSION THREAD

**Discussion by dlintw** [LazyVim/LazyVim#1494] [Q&A] [answered]
As https://github.com/LazyVim/LazyVim/discussions/1481, I know how to customize my snippets.
But I want to replace some "friendly-snippets" provided snippets to mine.
For example, in shell.json, I want to replace the "read" as auto typing "read -rp" instead of "read -r".
After my previous custom setting, the "friendly-snippets" will get the first priority.

I've read https://github.com/L3MON4D3/LuaSnip/blob/master/DOC.md#loaders, but ssems the "override_prioity" not workable. 
Here is my updated ~/.config/nvim/lua/plugins/luasnip.lua

```
return {
	{
		"L3MON4D3/LuaSnip",

		config = function(_, opts)
			require("luasnip.loaders.from_vscode").lazy_load({
				paths = { "~/.config/nvim/snippets" },
			})
			opts.override_priority = 2000
			-- require("luasnip.loaders.from_lua").load({ paths = "~/.config/nvim/snippets" })
		end,
	},
}
```
~/.config/nvim/snippet/package.json
```
{
  "contributes": {
    "snippets": [
      {
        "language": ["shellscript", "shell", "sh", "zsh"],
        "path": "./shell.json"
      }
    ]
  }
}
```
~/.config/nvim/snippet/shell.json, my modification of is "read -rp" instead of "read -r"
```
{
    "read": {
        "prefix": "read",
        "body": "read -rp ${0:VAR}",
        "description": "Read input of ${VAR}."
    }
}
```

**Comment by dlintw** [comment]
My workaround is to reduce the friendly-snippets priority as adding a file ~/.config/nvim/lua/config/friendly-snippets.lua, but this method still display two "read" option when I type "read" in a foo.sh file.
```
return {
	{
		"rafamadriz/friendly-snippets",
		config = function()
			require("luasnip.loaders.from_vscode").lazy_load()
		end,
		priority = 500,
	},
}
```

**Reply by dpetka2001** [reply]
You get duplicate `read` option, because you get completions from both `friendly-snippets` and your own custom snippets. I'm not sure if you can completely replace one of the default snippets with one of your own. The solution I see is to provide a different trigger key in your custom snippets to differentiate between the 2 or completely copy/paste the default snippets in your file and then exclude from `friendly-snippets` the language that you're trying to make snippets. You can see an example for the latter in the official [docs](https://github.com/rafamadriz/friendly-snippets#excluding-snippets).

**Comment by dlintw** [comment, marked as answer]
Here is my workable ~/.config/nvim/lua/plugins/luasnip.lua  which is a 2 lines modified edition from the default config of lazyvim.
```
return {
  "L3MON4D3/LuaSnip",
  build = (not jit.os:find("Windows"))
      and "echo 'NOTE: jsregexp is optional, so not a big deal if it fails to build'; make install_jsregexp"
    or nil,
  config = function()
    require("luasnip.loaders.from_vscode").lazy_load({
      paths = { "~/.config/nvim/snippets" },    -- MY CUSTOM snippets PATH
    })
  end,
  dependencies = {
    "rafamadriz/friendly-snippets",
    config = function()
      require("luasnip.loaders.from_vscode").lazy_load({
        exclude = { "sh" },  -- ADD THIS LINE
      })
    end,
  },
  opts = {
    history = true,
    delete_check_events = "TextChanged",
  },
  -- stylua: ignore
  keys = {
    {
      "<tab>",
      function()
        return require("luasnip").jumpable(1) and "<Plug>luasnip-jump-next" or "<tab>"
      end,
      expr = true, silent = true, mode = "i",
    },
    { "<tab>", function() require("luasnip").jump(1) end, mode = "s" },
    { "<s-tab>", function() require("luasnip").jump(-1) end, mode = { "i", "s" } },
  },
}
```
